core-primitives/node-api/metadata-provider/src/lib.rs

```diff
@@ -86,32 +85,3 @@ where
 		}
 	}
 }
-
-#[cfg(test)]
-mod tests {
-	use super::*;
-	use std::assert_matches::assert_matches;
-
-	#[derive(Default)]
-	struct NodeMetadataMock;
-
-	impl NodeMetadataMock {
-		fn get_one(&self) -> u32 {
-			1
-		}
-	}
-	#[test]
-	fn get_from_meta_data_returns_error_if_not_set() {
-		let repo = NodeMetadataRepository::<NodeMetadataMock>::default();
-
-		assert_matches!(repo.get_from_metadata(|m| m.get_one()), Err(Error::MetadataNotSet));
-	}
-
-	#[test]
-	fn get_from_metadata_works() {
-		let repo = NodeMetadataRepository::<NodeMetadataMock>::default();
-		repo.set_metadata(NodeMetadataMock);
-
-		assert_eq!(1, repo.get_from_metadata(|m| m.get_one()).unwrap());
-	}
-}
```

Comment:
Can we maybe add this again now?

Reply:
> Can we maybe add this again now?

Sure, just [added it](https://github.com/integritee-network/worker/pull/1148/commits/1d14d77752f2d42b04218bc59f770ec745e04a6b). To do it I had to loosen the trait bound, which I think is OK: it's fine to have the `get_from_metadata` interface for a wider type of T, but in the real logic we expect `MetadataType` to be bound by `NodeMetadataTrait`